libbeat/common/schema/mapstriface/mapstriface.go

```diff
@@ -103,7 +103,8 @@ func (convMap ConvMap) Map(key string, event mapstr.M, data map[string]interface
 	switch subData := d.(type) {
 	case map[string]interface{}, mapstr.M:
 		subEvent := mapstr.M{}
-		_, errs := convMap.Schema.ApplyTo(subEvent, subData.(map[string]interface{}))
+		convertedSubData, _ := subData.(map[string]interface{})
```

Comment:
We should check if the type assertion succeeded. Otherwise, if subData isn’t the right type, it panics.

Reply:
I'm not sure we need to worry about conversion failing here. 

For context, I made this change in response to a linter error that was triggered by an unrelated modification I made to this file. The original code didn't perform any error checking on this so I didn't introduce any myself. I assume it's because the `case` is specifically handling conversion of either `map[string]interface{}` or `mapstr.M`, which is basically just a wrapper around `map[string]interface{}`.

I'd be happy to add some error checking around the conversion if we think there's still a chance it could fail. What are your thoughts?